CORE-1450 Update to model v5.2

psarando opened the pull request:
This PR will update the `model` package to `v5.2` with the fix from cyverse-de/model#20.

This required updating the import paths `gopkg.in/cyverse-de/model` and `gopkg.in/cyverse-de/messaging` to `github.com/cyverse-de/model` and `github.com/cyverse-de/messaging` for `go mod`.

psarando committed:
CORE-1450 Update to model v5.2

slr71 reviewed (changes requested):
I don't think this is going to work the way it is, unfortunately. Switching to Go modules changes the import path from `gopkg.in/cyverse-de/model.v5` to `github.com/cyverse-de/model`, which will require us to change the import path. Unfortunately, that's going to require us to change other repositories as well because the import path has to be updated in other dependencies of `road-runner` that also use `model`. So far, I think that `model` and `messaging` are the only two that we'll have to update.

slr71 commented:
Oh, one thing I meant to mention in my review comment was that I haven't been able to find a way to get `go get` to fully update a dependency from `gopkg.in`. If we could get that to work then we could avoid sweeping changes.

psarando committed:
CORE-1450 Update model and messaging imports

Update imports from gopkg.in to github.com
Also update dcompose test with more step params.

psarando commented on status.go:
```diff
@@ -3,10 +3,17 @@ package main
+	"github.com/cyverse-de/model"
 )
 
+func jobDetailsFromJob(job *model.Job) messaging.JobDetails {
+	return messaging.JobDetails{
+		InvocationID: job.InvocationID,
+		CondorID:     job.CondorID,
+	}
+}
```
Thanks to @slr71 for this suggestion 👍

slr71 reviewed (approved):
Looks good to me! 👍 :shipit:

psarando commented:
Thanks for the review! 👍
